planning/behavior_velocity_dynamic_obstacle_stop_module/src/collision.cpp

```diff
@@ -63,13 +66,23 @@ std::optional<geometry_msgs::msg::Point> find_closest_collision_point(
 std::vector<Collision> find_collisions(
   const EgoData & ego_data,
   const std::vector<autoware_auto_perception_msgs::msg::PredictedObject> & objects,
-  const tier4_autoware_utils::MultiPolygon2d & object_forward_footprints)
+  const tier4_autoware_utils::MultiPolygon2d & object_forward_footprints,
+  const PlannerParam & params)
 {
   std::vector<Collision> collisions;
+  std::optional<geometry_msgs::msg::Point> collision;
   for (auto object_idx = 0UL; object_idx < objects.size(); ++object_idx) {
     const auto & object_pose = objects[object_idx].kinematics.initial_pose_with_covariance.pose;
-    const auto & object_footprint = object_forward_footprints[object_idx];
-    const auto collision = find_closest_collision_point(ego_data, object_pose, object_footprint);
+    if (!params.ignore_objects_behind_ego) {
+      tier4_autoware_utils::MultiPolygon2d object_footprint;
+      for (const auto & polygon : object_forward_footprints) {
+        object_footprint.push_back(polygon);
+        collision = find_closest_collision_point(ego_data, object_pose, polygon, params);
```

Comment:
I think there is a problem in this loop. Only the `collision` calculated for the last `polygon` will be used and the `object_footprint` is not used.

Do you think we can change `object_forward_footprints` to be a `std::vector<tier4_autoware_utils::MultiPolygon2d> object_footprints` such that in both cases (multiple polygons generated from a predicted path or one "forward" polygon) we always have `object_footprints[i]` corresponding to the object of index `i` ? Then the same collision checking code can be used for both cases.